src/geneweaver/aon/controller/genes.py

```diff
@@ -22,13 +34,32 @@ def get_gene(gene_id: int, db: deps.Session = Depends(deps.session)):
     return genes_service.gene_by_id(gene_id, db)
 
 
-@router.get("/ref_ids/{ref_id}")
-def get_gene_by_ref_id(ref_id: str, db: deps.Session = Depends(deps.session)):
+@router.get("/by-ref-id/{ref_id}")
+def get_gene_by_ref_id(
+    ref_id: str,
+    ref_id_type: ReferenceGeneIDType = ReferenceGeneIDType.GW,
+    db: deps.Session = Depends(deps.session),
+):
     """Get gene by reference id."""
+    if ref_id_type == ReferenceGeneType.AON_REF_ID:
+        ref_id = convert_service.ode_ref_to_agr(db, ref_id)
     return genes_service.gene_by_ref_id(ref_id, db)
 
 
-@router.get("/prefixes")
-def get_gene_prefixes(db: deps.Session = Depends(deps.session)):
-    """Get gene prefixes."""
-    return genes_service.gene_prefixes(db)
+@router.get("/covert/{id}")
+def covert_gene_id(
```

Comment:
convert_gene_id instead of covert_gene_id?

Reply:
I'll fix this in the next PR.